zinc/src/sbt-test/source-dependencies/less-inter-inv-java/test

```diff
@@ -6,4 +6,4 @@ $ copy-file changes/A2.java A.java
 # 1 iteration to recompile all descendents and direct dependencies
 # no further iteration, because APIs of directs don't change
 > run
-> checkIterations 2
+> checkIterations 1
```

Comment:
Could you update the comment?  Same with where we disable pipelining (including source-dependencies/subproject-dependency) could you leave a note with some reasoning, please?

Reply:
In general Zinc's scripted test are like do X1, X2, ..., `compile`, and see if `A` was recompiled as a result, or not.
Pipelining itself is ok, but the `compileAllJava(...)` at the end messes this up because all Java classes are invalidated. This _reduces_ the fidelity of our test because we're essentially over-compiling all the time. I turned off `pipelining` due to that limitation.

You or others can/should try to work out smaller sets of Java sources to compile that retains the same characteristic of incrementality (by tracking list of Java sources that needs recompilation in Analysis etc).

I'd have to look more into the `less-inter-inv-java`.

Reply:
There was a regression in https://github.com/sbt/zinc/pull/744 for non-pipelined cases that's losing some Analysis data.

Reply:
Regression is addressed in https://github.com/sbt/zinc/pull/843/commits/62e4432024f3a60e356a255f531d5822089d2393.

Reply:
I think that's the one that was breaking the "Created a Compilations with duplicate compilations" in `MCompilations`.

But I guess we'll see what CI thinks.